libs/langchain/langchain/document_loaders/pdf.py

```diff
@@ -440,3 +476,127 @@ def load(self) -> List[Document]:
+            )
+            document = loader.load()
+    """
+
+    try:
+        import textractcaller as tc
+    except ImportError:
+        raise ModuleNotFoundError(
+            "Could not import amazon-textract-caller python package. \
+            Please install it with `pip install amazon-textract-caller` \
+            or add to your requirements.txt."
+        )
+
+    def __init__(
+        self,
+        file_path: str,
+        textract_features: List[tc.Textract_Features] = list(),
+        client: Any = None,
+        credentials_profile_name: str = "",
+        region_name: str = "",
+        endpoint_url: str = "",
+    ) -> None:
+        super().__init__(file_path)
+        if credentials_profile_name or region_name or endpoint_url:
+            try:
+                import boto3
+
+                if credentials_profile_name is not None:
+                    session = boto3.Session(profile_name=credentials_profile_name)
+                else:
+                    # use default credentials
+                    session = boto3.Session()
+
+                client_params = {}
+                if region_name:
+                    client_params["region_name"] = region_name
+                if endpoint_url:
+                    client_params["endpoint_url"] = endpoint_url
+
+                client = session.client("textract", **client_params)
+
+            except ImportError:
+                raise ModuleNotFoundError(
+                    "Could not import boto3 python package. "
+                    "Please install it with `pip install boto3`."
+                )
+            except Exception as e:
+                raise ValueError(
+                    "Could not load credentials to authenticate with AWS client. "
+                    "Please check that credentials in the specified "
+                    "profile name are valid."
+                ) from e
+        self.parser = AmazonTextractPDFParser(
+            textract_features=textract_features, client=client
+        )
+
+    def load(self) -> List[Document]:
+        """Load given path as pages."""
+        return list(self.lazy_load())
+
+    def lazy_load(
+        self,
+    ) -> Iterator[Document]:
+        """Lazy load documents"""
+        # the self.file_path is local, but the blob has to include
+        # the S3 location if the file originated from S3 for multi-page documents
+        # raises ValueError when multi-page and not on S3"""
+        blob = Blob.from_path(self.file_path)
+        if AmazonTextractPDFLoader._get_number_of_pages(blob) > 1:
+            if self.web_path and self._is_s3_url(self.web_path):
+                blob = Blob(data=None, mimetype=blob.mimetype, path=self.web_path)
+            else:
+                raise ValueError(
+                    f"the file {blob.path} is a multi-page document, \
+                    but not stored on S3. \
+                    Textract requires multi-page documents to be on S3."
```

Comment:
(not for this pr) We don't have an s3 blob generator yet, if you all are interested in contributing -- it'll create a convenient interface for users to use the parser together with content they have stored on s3.

----

i assume this is a restriction of the API itself?